merino-integration-tests/src/caching/redis_tests.rs

```diff
@@ -197,3 +198,92 @@ async fn get_cached_key(
     })
     .await
 }
+
+#[merino_test_macro(|settings| {
+    settings.debug = true;
+    settings.suggestion_providers.insert(
+        "fixed_redis".to_string(),
+        SuggestionProviderConfig::RedisCache(RedisCacheConfig::with_inner(
+            SuggestionProviderConfig::Fixed(FixedConfig { value: "foo".to_owned() })
+        )),
+    );
+})]
+async fn test_reconfigure(
+    TestingTools {
+        test_client,
+        mut redis_client,
+        ..
+    }: TestingTools,
+) {
+    let keys_before: Vec<String> = redis_client
+        .keys("provider:v1:*")
+        .expect("Could not get keys from redis");
+    assert!(keys_before.is_empty());
+
+    let response = test_client
+        .get("/api/v1/suggest?q=foo")
+        .send()
+        .await
+        .expect("failed to execute request");
+
+    assert_eq!(response.status(), StatusCode::OK);
+
+    // Prepare for updating the inner provider.
+    // Reset the inner `FixedProvider` also reduce the `default_ttl`.
+    let mut redis_config =
+        RedisCacheConfig::with_inner(SuggestionProviderConfig::Fixed(FixedConfig {
+            value: "bar".to_owned(),
+        }));
+    redis_config.default_ttl = Duration::from_secs(1);
+    let new_config: HashMap<String, SuggestionProviderConfig> = [(
+        "fixed_redis".to_owned(),
+        SuggestionProviderConfig::RedisCache(redis_config),
+    )]
+    .into_iter()
+    .collect();
+
+    // Request for a reconfigure for Redis provider.
+    // Note that this reconfigure endpoint is only exposed in debug mode.
+    let response = test_client
+        .post("/api/v1/providers/reconfigure")
+        .json(&new_config)
+        .send()
+        .await
+        .expect("failed to execute request");
+    assert_eq!(response.status(), StatusCode::NO_CONTENT);
+
+    let response = test_client
+        .get("/api/v1/suggest?q=bar")
+        .send()
+        .await
+        .expect("failed to execute request");
+    assert_eq!(response.status(), StatusCode::OK);
+
+    // Verify the new `FixedProvider` is in use.
+    let http_response: Value = response.json().await.expect("response was not json");
+    let http_suggestions = http_response["suggestions"].as_array().unwrap();
+    assert_eq!(http_suggestions.len(), 1);
+    assert_eq!(http_suggestions[0]["title"], "bar");
+
+    // Wait for a bit to let the newly inserted value expire.
+    tokio::time::sleep(Duration::from_millis(1500)).await;
```

Comment:
Eugh :( But also I don't think there's any other way around this

Reply:
Lol, say Ciao to Mr. Sanfilippo for me, Grazie!